File: jobs/2026-06-08__12-44-45/cron_job_cleanup_ts__k5FRLvU/artifacts/user/cron-app-zr-k5frlvu/cleanup/cleanup.ts
import { api, rawApi } from "encore.dev/api";
import { SQLDatabase } from "encore.dev/storage/sqldb";
import { CronJob } from "encore.dev/cron";

// Define the database
const db = new SQLDatabase("records_db", {
  migrations: "./migrations",
});

interface DbRecord {
  id: number;
  data: string;
  created_at: string;
}

interface CreateRecordParams {
  data: string;
  created_at?: string;
}

interface CreateRecordResponse {
  id: number;
}

// POST /records
export const createRecord = api(
  { expose: true, method: "POST", path: "/records" },
  async (params: CreateRecordParams): Promise<CreateRecordResponse> => {
    const { data, created_at } = params;
    let row;
    if (created_at) {
      row = await db.queryRow`
        INSERT INTO records (data, created_at)
        VALUES (${data}, ${created_at})
        RETURNING id
      `;
    } else {
      row = await db.queryRow`
        INSERT INTO records (data)
        VALUES (${data})
        RETURNING id
      `;
    }
    if (!row) throw new Error("Failed to insert record");
    return { id: row.id as number };
  }
);

// GET /records
export const getRecords = rawApi(
  { expose: true, method: "GET", path: "/records" },
  async (req, resp) => {
    const rows = db.query`
      SELECT id, data, created_at
      FROM records
      ORDER BY id
    `;
    const records: DbRecord[] = [];
    for await (const row of rows) {
      records.push({
        id: row.id as number,
        data: row.data as string,
        created_at: (row.created_at as Date).toISOString(),
      });
    }
    resp.setHeader("Content-Type", "application/json");
    resp.end(JSON.stringify(records));
  }
);

interface CleanupResponse {
  deleted_count: number;
}

// POST /cleanup
export const cleanup = api(
  { expose: true, method: "POST", path: "/cleanup" },
  async (): Promise<CleanupResponse> => {
    const oneHourAgo = new Date(Date.now() - 3600000);
    const result = await db.exec`
      DELETE FROM records
      WHERE created_at < ${oneHourAgo}
    `;
    return { deleted_count: Number(result.rowsAffected) };
  }
);

// Cron Job: cleanup-job every 1 hour
const _ = new CronJob("cleanup-job", {
  every: "1h",
  endpoint: cleanup,
});

File: jobs/2026-05-30__07-57-35/cron_job_cleanup_go__qMCgLXY/artifacts/user/cleanup-app/cleanup/cleanup.go
package cleanup

import (
	"context"
	"time"

	"encore.dev/cron"
	"encore.dev/storage/sqldb"
)

// Define the database
var cleanupDB = sqldb.NewDatabase("cleanup_db", sqldb.DatabaseConfig{
	Migrations: "./migrations",
})

// Request/Response types
type Record struct {
	ID        string `json:"id"`
	Data      string `json:"data"`
	CreatedAt string `json:"created_at"`
}

type InsertRequest struct {
	ID        string `json:"id"`
	Data      string `json:"data"`
	CreatedAt string `json:"created_at"`
}

type ListResponse struct {
	Records []Record `json:"records"`
}

type CleanupResponse struct {
	DeletedCount int `json:"deleted_count"`
}

//encore:api public method=POST path=/records
func InsertRecord(ctx context.Context, req *InsertRequest) error {
	_, err := cleanupDB.Exec(ctx, `
		INSERT INTO records (id, data, created_at)
		VALUES ($1, $2, $3)
	`, req.ID, req.Data, req.CreatedAt)
	return err
}

//encore:api public method=GET path=/records
func ListRecords(ctx context.Context) (*ListResponse, error) {
	rows, err := cleanupDB.Query(ctx, `
		SELECT id, data, created_at FROM records
	`)
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	var records []Record
	for rows.Next() {
		var r Record
		var createdAt time.Time
		if err := rows.Scan(&r.ID, &r.Data, &createdAt); err != nil {
			return nil, err
		}
		r.CreatedAt = createdAt.Format(time.RFC3339)
		records = append(records, r)
	}
	if records == nil {
		records = []Record{}
	}
	return &ListResponse{Records: records}, nil
}

//encore:api public method=POST path=/cleanup
func Cleanup(ctx context.Context) (*CleanupResponse, error) {
	cutoff := time.Now().Add(-24 * time.Hour)
	result, err := cleanupDB.Exec(ctx, `
		DELETE FROM records WHERE created_at < $1
	`, cutoff)
	if err != nil {
		return nil, err
	}
	affected, _ := result.RowsAffected()
	return &CleanupResponse{DeletedCount: int(affected)}, nil
}

// Cron job that runs every hour to trigger cleanup
var _ = cron.NewJob("cleanup-cron", cron.JobConfig{
	Every:    1 * cron.Hour,
	Endpoint: Cleanup,
})